tests/integration/aws_s3/test_aws_s3.py

```diff
@@ -60,73 +57,101 @@ def aws_s3_configured(extra_params=None):
+        cls._fs = fsspec.filesystem(
+            "s3", anon=False, key=cls._access_key_id, secret=cls._secret_access_key
         )
-        profile = DatastoreProfileS3(
-            name="s3ds_profile",
+
+    @classmethod
+    def teardown_class(cls):
+        test_dir = f"{cls._bucket_name}{cls.test_dir}"
+        if not cls._fs:
+            return
+        if cls._fs.exists(test_dir):
+            cls._fs.delete(test_dir, recursive=True)
+            logger.debug("test directory has been deleted.")
+
+    def setup_method(self, method):
+        store_manager.reset_secrets()
+        self.profile = DatastoreProfileS3(
+            name=self.profile_name,
             access_key_id=self._access_key_id,
             secret_key=self._secret_access_key,
         )
-        register_temporary_client_datastore_profile(profile)
+        register_temporary_client_datastore_profile(self.profile)
+
+    @pytest.fixture(autouse=True)
+    def setup_before_each_test(self, use_datastore_profile):
+        # We give priority to profiles, then to secrets, and finally to environment variables.
+        # We want to ensure that we test these priorities in the correct order.
+        mlrun.datastore.store_manager.reset_secrets()
+
+        if use_datastore_profile:
+            os.environ["AWS_ACCESS_KEY_ID"] = "wrong_access_key"
+            os.environ["AWS_SECRET_ACCESS_KEY"] = "wrong_token"
+            prefix_path = f"ds://{self.profile_name}/"
+        else:
+            os.environ["AWS_ACCESS_KEY_ID"] = self._access_key_id
+            os.environ["AWS_SECRET_ACCESS_KEY"] = self._secret_access_key
+            prefix_path = "s3://"
+        self._bucket_path = f"{prefix_path}{self._bucket_name}"
+        self.run_dir_url = f"{self._bucket_path}{self.run_dir}"
+        object_file = f"/file_{uuid.uuid4()}.txt"
+        self._object_url = self.run_dir_url + object_file
 
     def _perform_aws_s3_tests(self, use_datastore_profile, secrets=None):
-        param = self.s3["ds"] if use_datastore_profile else self.s3["s3"]
-        logger.info(f'Object URL: {param["object_url"]}')
+        #  TODO split to smaller tests, according to datastore's tests convention.
+        logger.info(f"Object URL: {self._object_url}")
 
-        data_item = mlrun.run.get_dataitem(param["object_url"], secrets=secrets)
-        data_item.put(test_string)
-        df_data_item = mlrun.run.get_dataitem(param["df_url"], secrets=secrets)
+        data_item = mlrun.run.get_dataitem(self._object_url, secrets=secrets)
+        data_item.put(self.test_string)
+        df_url = f"{self.run_dir_url}/df_{uuid.uuid4()}.csv"
+        df_data_item = mlrun.run.get_dataitem(df_url, secrets=secrets)
         df_data_item.put(test_df_string)
 
         response = data_item.get()
-        assert response.decode() == test_string, "Result differs from original test"
+        assert (
+            response.decode() == self.test_string
+        ), "Result differs from original test"
```

Comment:
```suggestion
        )
```
I know it was there before, but this assertion message is more confusing than helpful.

Naturally this refers to all occurrences, not just this line.